### PR TITLE
Upgrade the `form-data` lib

### PR DESCRIPTION
**Pull request by jkoenig134**
# Readiness checklist

-   [ ] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description

As described in https://github.com/form-data/form-data/issues/563 we want to make sure to be ready for node 23. As `form-data` 4.0.1 is now released this should do the job.

### CODE REVIEW

File: packages/transport/src/core/TransportCoreErrors.ts [hunk 1]
```diff
@@ -104,20 +104,13 @@ class Challenges {
 }
 
 class Datawallet {
-    public unsupportedModification(type: "unsupportedCacheChangedModificationCollection", data: any) {
-        const errorCode = "error.transport.datawallet.unsupportedModification";
+    public cacheChangeForUnsupportedCollectionDetected(data: any) {
```
Comment (Magnus-Kuhn) [inline packages/transport/src/core/TransportCoreErrors.ts:107]: This renaming technically constitutes a breaking change and the code is included in the documentation, is it worth it?

Comment (jkoenig134) [reply packages/transport/src/core/TransportCoreErrors.ts:107]: Docs yes, breaking no. I doubt that anyone relies on it.

Comment (jkoenig134) [reply packages/transport/src/core/TransportCoreErrors.ts:107]: I think I have a solution to make it really backwards compatible.

Comment (jkoenig134) [reply packages/transport/src/core/TransportCoreErrors.ts]: @Magnus-Kuhn changed, thanks for the heads-up!

**Review by Magnus-Kuhn — APPROVED** [review]

**Pull request merged**